position/position_test.go

```diff
@@ -0,0 +1,76 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package position
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+	"github.com/matryer/is"
+)
+
+func TestParseSDKPosition(t *testing.T) {
+	validPosition := Position{
+		IteratorType:   TypeCDC,
+		Key:            "key",
+		SequenceNumber: "my-sequence-number",
+		Time:           time.Time{},
+	}
+
+	wrongPosType := Position{
+		IteratorType:   3, // non-existent type
+		Key:            "key",
+		SequenceNumber: "my-sequence-number",
+	}
+	is := is.New(t)
+	posBytes, err := json.Marshal(validPosition)
+	is.NoErr(err)
+
+	wrongPosBytes, err := json.Marshal(wrongPosType)
+	is.NoErr(err)
+
+	tests := []struct {
+		name        string
+		in          opencdc.Position
+		want        Position
+		expectedErr string
+	}{
+		{
```

Comment:
Maybe a test for a nil or an empty position would be good too, because that's what's given to a connector the first time it starts?

Reply:
done